packages/core/src/services/scales-cartesian.ts

```diff
@@ -214,6 +214,15 @@ export class CartesianScales extends Service {
 		}
 	}
 
+	getCustomDomainValuesByposition(axisPosition: AxisPositions) {
+		return Tools.getProperty(
+			this.model.getOptions(),
+			'axes',
+			axisPosition,
+			'domain'
+		);
+	}
```

Comment:
let's add that validation here... isArray && length === 2, and we can probably send a console.error if the wrong format is provided

Reply:
Fixed:

```ts
	getCustomDomainValuesByposition(axisPosition: AxisPositions) {
		const domain = Tools.getProperty(
			this.model.getOptions(),
			'axes',
			axisPosition,
			'domain'
		);

		// Check if domain is an array
		if (domain && !Array.isArray(domain)) {
			throw new Error(
				`Domain in ${axisPosition} axis is not a valid array`
			);
		}

		// Determine number of elements passed in domain depending on scale types
		if (Array.isArray(domain)) {
			if (
				(this.scaleTypes[axisPosition] === ScaleTypes.LINEAR ||
					this.scaleTypes[axisPosition] === ScaleTypes.TIME) &&
				domain.length !== 2
			) {
				throw new Error(
					`There can only be 2 elements in domain for scale type: ${this.scaleTypes[axisPosition]}`
				);
			}
		}

		return domain;
	}
```